daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/ledger/BlindingTransaction.scala

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.ledger
+
+import com.daml.lf.data.Ref.Party
+import com.daml.lf.data.Relation.Relation
+import com.daml.lf.transaction.BlindingInfo
+import com.daml.lf.transaction.Node.{NodeCreate, NodeExercises, NodeFetch, NodeLookupByKey}
+import com.daml.lf.transaction.{NodeId, Transaction => Tx}
+import com.daml.lf.value.Value.ContractId
+
+object BlindingTransaction {
+
+  def crash(reason: String) =
+    throw new IllegalArgumentException(reason)
+
+  private object BlindState {
+    def Empty =
```

Comment:
```suggestion
    val Empty =
```
Any reason to create a new one every time this is called?

Reply:
Nope